pkg/datapath/linux/ipsec/ipsec_linux.go

```diff
@@ -289,42 +298,57 @@ func decodeIPSecKey(keyRaw string) ([]byte, error) {
 
 // LoadIPSecKeysFile imports IPSec auth and crypt keys from a file. The format
 // is to put a key per line as follows, (auth-algo auth-key enc-algo enc-key)
-func LoadIPSecKeysFile(path string) error {
+func LoadIPSecKeysFile(path string) (uint8, error) {
 	file, err := os.Open(path)
 	if err != nil {
-		return err
+		return 0, err
 	}
 	defer file.Close()
 	return loadIPSecKeys(file)
 }
 
-func loadIPSecKeys(r io.Reader) error {
+func loadIPSecKeys(r io.Reader) (uint8, error) {
```

Comment:
nit: For future review, it'd be nice to place an example input above here. Bit tricky to really look this over and understand the string manipulation below without an example. I see associated unit tests so not so worried, could be a minor follow-up comment in another commit/PR.

Reply:
I've got a set of cleanups/pretty comments I'll add it to those.